### PR TITLE
OSD-6949 - Add test for custom domains operator.

### PR DESCRIPTION
**Pull request by tnierman**
Adds test described at https://github.com/openshift/custom-domains-operator/blob/master/TESTING.md#live-testing to ensure custom domains operator is functioning as intended.

### CODE REVIEW

File: pkg/e2e/operators/customdomains.go [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/rest"
+)
+
```
Comment (dustman9000) [inline pkg/e2e/operators/customdomains.go:38]: nit: can we reduce the number of double line endings in this file? grouping logical lines together makes it easier to read.

File: pkg/e2e/operators/customdomains.go [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+		ginkgo.BeforeEach(func(){
+			ginkgo.By("Logging in as a dedicated-admin")
+
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "dummy-admin@redhat.com",
+				Groups: []string{
+					"dedicated-admins",
+				},
+
```
Comment (dustman9000) [inline pkg/e2e/operators/customdomains.go:70]:
```suggestion
```

File: pkg/e2e/operators/customdomains.go [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+		ginkgo.It("Should be resolvable by external services", func() {
+			ginkgo.By("Logging in as a dedicated-admin")
+
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "dummy-admin@redhat.com",
+				Groups: []string{
+					"dedicated-admins",
+				},
+
```
Comment (dustman9000) [inline pkg/e2e/operators/customdomains.go:199]:
```suggestion
```

File: pkg/e2e/operators/customdomains.go [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+					Ports: []corev1.ServicePort{
+						{
+							Name: "8080-tcp",
+							Protocol: corev1.ProtocolTCP,
+							Port: 8080,
+							TargetPort: intstr.IntOrString{
+								Type: intstr.Int,
+								IntVal: 8080,
+							},
+
```
Comment (dustman9000) [inline pkg/e2e/operators/customdomains.go:274]:
```suggestion
```

File: pkg/e2e/operators/customdomains.go [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+					},
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationEdge,
+					},
+					To: routev1.RouteTargetReference{
+						Kind: "Service",
+						Name: testAppService.GetName(),
+					},
+				},
+
```
Comment (dustman9000) [inline pkg/e2e/operators/customdomains.go:316]:
```suggestion
```

File: pkg/e2e/operators/customdomains.go [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+						if addr == testRoute.Spec.Host + ":443"{
+							addr = testDomain.Status.Endpoint + ":443"
+						}
+						return dialer.DialContext(context, network, addr)
+					},
+				},
+			}
+
+			var response *http.Response
+			wait.PollImmediate(30*time.Second,15*time.Minute,func() (bool, error) {
```
Comment (dustman9000) [inline pkg/e2e/operators/customdomains.go:343]: Can we move these numbers into a const section?

**Comment by dustman9000** [comment]
This is really great, thanks for putting this together! Just some minor cosmetic comments, overall looks good for the basic functionality test.
/lgtm

File: pkg/e2e/operators/customdomains.go (hunk 1 above)
Comment (tnierman) [reply pkg/e2e/operators/customdomains.go:38]: I reorganized some of the logic and removed the extraneous spaces you pointed out in other comments. Let me know if you think this is sufficient, or if there are more change you'd like to see.

Comment (dustman9000) [reply pkg/e2e/operators/customdomains.go:38]: Looks good, thanks!

**Review by whereswaldon — APPROVED** [review]
/approve

Thanks! Looks solid.

**Pull request merged**